### PR TITLE
refactor: update lints to embark lints 0.3

### PR DESCRIPTION
**Pull request by DeanBDean**
### Checklist

* [x] I have read the [Contributor Guide](../../CONTRIBUTING.md)
* [x] I have read and agree to the [Code of Conduct](../../CODE_OF_CONDUCT.md)
* [x] I have added a description of my changes and why I'd like them included in the section below

### Description of Changes

Updated linting rules to Embark 0.3 and made minor changes to satisfy the linter

### Related Issues

https://github.com/EmbarkStudios/puffin/issues/19

### CODE REVIEW

File: puffin-imgui/src/ui.rs [hunk 1]
```diff
@@ -153,9 +153,10 @@ impl<'ui> Painter<'ui> {
     }
 
     fn pixel_from_ns(&self, options: &Options, ns: NanoSecond) -> f32 {
-        self.canvas_min.x
-            + options.sideways_pan_in_pixels
-            + ((ns - options.start_ns) as f32) * options.pixels_per_ns
+        ((ns - options.start_ns) as f32).mul_add(
+            options.pixels_per_ns,
+            self.canvas_min.x + options.sideways_pan_in_pixels,
+        )
```
Comment (emilk) [inline puffin-imgui/src/ui.rs:159]: Less parenthesis:

```suggestion
        options.pixels_per_ns.mul_add(
            (ns - options.start_ns) as f32,
            self.canvas_min.x + options.sideways_pan_in_pixels,
        )
```

File: puffin-imgui/src/ui.rs [hunk 2]
```diff
@@ -356,8 +357,9 @@ impl ProfilerUi {
 
             self.options.pixels_per_ns *= zoom_factor;
             let zoom_center = painter.mouse_pos.x - painter.canvas_min.x;
-            self.options.sideways_pan_in_pixels =
-                (self.options.sideways_pan_in_pixels - zoom_center) * zoom_factor + zoom_center;
+            self.options.sideways_pan_in_pixels = (self.options.sideways_pan_in_pixels
+                - zoom_center)
+                .mul_add(zoom_factor, zoom_center);
```
Comment (emilk) [inline puffin-imgui/src/ui.rs:362]: Fewer parenthesis (again):

```suggestion
            self.options.sideways_pan_in_pixels = zoom_factor.mul_add(
                    self.options.sideways_pan_in_pixels - zoom_center, zoom_center);
```

File: puffin-imgui/src/ui.rs [hunk 3]
```diff
@@ -526,7 +528,7 @@ fn paint_record(
                 painter.draw_list.add_text(
                     [
                         start_x + 4.0,
-                        top_y + 0.5 * (options.rect_height - painter.font_size),
+                        0.5f32.mul_add(options.rect_height - painter.font_size, top_y),
```
Comment (emilk) [inline puffin-imgui/src/ui.rs:531]: ugh... I am not a big fan of `clippy::suboptimal_flops`. It really hurts readability IMHO :/
Maybe we could just silence the warning for this file with `#![allow(clippy::suboptimal_flops)]` instead and revert this particular line. Or maybe I just need to get used to reading code with `mul_add` :)

**Review by emilk — CHANGES_REQUESTED** [review]
Thank you for this PR!

Some smaller comments about the `suboptimal_flops`-suggested changes, but otherwise LGTM!

File: puffin-imgui/src/ui.rs (hunk 3 above)
Comment (DeanBDean) [reply puffin-imgui/src/ui.rs:531]: So, I believe this lint is just an optimization, but if you don't need the optimization then to me readability should win out (I think the other way is more readable). I don't know the code well enough to say, although I do suspect it's not needed. Either way, let me know how you want to go and I'll implement it.

Comment (emilk) [reply puffin-imgui/src/ui.rs:531]: I'd like to have `#![allow(clippy::suboptimal_flops)]` at the module level and revert these changes please :)

**Comment by DeanBDean** [comment]
@emilk I reverted the suboptimal flops back and disabled that lint

**Review by emilk — APPROVED** [review]
Thank you!

**Pull request merged**